### ISSUE TITLE
💌 #FeatureRequest as a user, when I visit a user search, I want to see if a user is listed

### ISSUE THREAD

**Issue by ldexterldesign** [tomitm/lists#4]
Hey,

Thanks for extension

💌 #FeatureRequest as a user, when I visit a user search (e.g. https://twitter.com/search?f=users&q=bbc ), I want to see if a user is listed so I can quickly CRUD users to lists without visiting their profile

http://quick.as/6xnXHy5gk

If you agree reply +1 or vote this up

Please let me know if there's a more suitable place to leave issues?

Device: Apple MBP (desktop)
Platform: OS X
Version: 10.11.5
Browser: Google Chrome Version 50.0.2661.102 (64-bit)

Hope this is useful and to hear back

Kind thanks

**Comment by tomitm** [comment]
The challenge in determining whether a user is in a list or not is that I need to make an individual API request for each user to find which lists they're in.

I don't seem to be able to make a single bulk request for *n* users, nor does the list of lists API request (which I use on the sidebar) contain the members in each lists.

Making a bunch of separate requests would make my extension a pretty bad API consumer and would likely impact your browsing experience by affecting performance.

There is one tweak I can safely make, which is sorting the list of lists when you hover over a user or click add/remove from lists to show the checked lists first. Would that help?

Thanks for all the information and video!

**Comment by ldexterldesign** [comment]
Thanks for reply

Ahh, that's too bad, but see your logic

> There is one tweak I can safely make, which is sorting the list of lists when you hover over a user or click add/remove from lists to show the checked lists first. Would that help?

Hell yea! I'll created a separate issue now...

Hope to hear back

Kind thanks

**Comment by tomitm** [comment]
I've implemented this on the `sort` branch - checked memberships are now on the top of the list in profile cards.

Testing this now, if all goes well, will release in the next couple of days.

If you'd like to help test:
- Clone this repo, then checkout the `sort` branch
- `npm install`
- `npm run build-dist`
- In `chrome://extensions`
  - Enable developer mode
  - Click *Load unpacked extension*
  - Select the `app` directory from your checkout folder

**Comment by ldexterldesign** [comment]
Thanks for reply

```
Apple-MacBook-Pro:lists ldexterldesign$ sudo npm run build-dist
Password:

> lists@2.0.7 build-dist /Users/ldexterldesign/Desktop/lists
> rollup -c rollup.config.dist.js

Cannot find module '../config.json' from '/Users/ldexterldesign/Desktop/lists/src'
Error: Cannot find module '../config.json' from '/Users/ldexterldesign/Desktop/lists/src'
    at /Users/ldexterldesign/Desktop/lists/node_modules/rollup-plugin-node-resolve/node_modules/resolve/lib/async.js:55:21
    at load (/Users/ldexterldesign/Desktop/lists/node_modules/rollup-plugin-node-resolve/node_modules/resolve/lib/async.js:69:43)
    at onex (/Users/ldexterldesign/Desktop/lists/node_modules/rollup-plugin-node-resolve/node_modules/resolve/lib/async.js:92:31)
    at /Users/ldexterldesign/Desktop/lists/node_modules/rollup-plugin-node-resolve/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:95:15)
Type rollup --help for help, or visit https://github.com/rollup/rollup/wiki

npm ERR! Darwin 15.5.0
npm ERR! argv "node" "/usr/local/bin/npm" "run" "build-dist"
npm ERR! node v0.12.7
npm ERR! npm  v2.13.5
npm ERR! code ELIFECYCLE
npm ERR! lists@2.0.7 build-dist: `rollup -c rollup.config.dist.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the lists@2.0.7 build-dist script 'rollup -c rollup.config.dist.js'.
npm ERR! This is most likely a problem with the lists package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     rollup -c rollup.config.dist.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls lists
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/ldexterldesign/Desktop/lists/npm-debug.log
Apple-MacBook-Pro:lists ldexterldesign$ 
```

```
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/usr/local/bin/npm', 'run', 'build-dist' ]
2 info using npm@2.13.5
3 info using node@v0.12.7
4 verbose run-script [ 'prebuild-dist', 'build-dist', 'postbuild-dist' ]
5 info prebuild-dist lists@2.0.7
6 info build-dist lists@2.0.7
7 verbose unsafe-perm in lifecycle true
8 info lists@2.0.7 Failed to exec build-dist script
9 verbose stack Error: lists@2.0.7 build-dist: `rollup -c rollup.config.dist.js`
9 verbose stack Exit status 1
9 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:214:16)
9 verbose stack     at EventEmitter.emit (events.js:110:17)
9 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:24:14)
9 verbose stack     at ChildProcess.emit (events.js:110:17)
9 verbose stack     at maybeClose (child_process.js:1015:16)
9 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
10 verbose pkgid lists@2.0.7
11 verbose cwd /Users/ldexterldesign/Desktop/lists
12 error Darwin 15.5.0
13 error argv "node" "/usr/local/bin/npm" "run" "build-dist"
14 error node v0.12.7
15 error npm  v2.13.5
16 error code ELIFECYCLE
17 error lists@2.0.7 build-dist: `rollup -c rollup.config.dist.js`
17 error Exit status 1
18 error Failed at the lists@2.0.7 build-dist script 'rollup -c rollup.config.dist.js'.
18 error This is most likely a problem with the lists package,
18 error not with npm itself.
18 error Tell the author that this fails on your system:
18 error     rollup -c rollup.config.dist.js
18 error You can get their info via:
18 error     npm owner ls lists
18 error There is likely additional logging output above.
19 verbose exit [ 1, true ]
```
- file:///Users/ldexterldesign/Desktop/lists/npm-debug.log

Hope this is useful and to hear back

Kind thanks

**Comment by tomitm** [comment]
Oh right, forgot about that issue. I'll create a new issue for myself to fix it later.

Just released this addition as 2.1.0, you should be automatically updated shortly! 😄

**Comment by ldexterldesign** [comment]
Oh, congrats on the v2.1.0 release!

http://quick.as/GlPVizpno - I don't appear to be able to see any new functionality relating to this issue in my browser?

Can you clarify which new feature you've added (eager to use it)?

Yours hopefully

**Comment by tomitm** [comment]
Hey, sorry about that! I added the sort to the lists in profile hover cards (checked lists on top), but overlooked that menu. Figures that the profile search doesn't use hover cards.

Currently I've only applied styling to that add/remove modal, but I'll see if I can detect when it opens so I can sort the list as well.

**Comment by tomitm** [comment]
Alright, got that modal working too! Fortunately very straight-forward as I had most things in place already, just a matter of reuse for the modal.

Going to give this a quick test, then push it out either tonight or tomorrow. 🚀

**Comment by tomitm** [comment]
Just released 2.1.1 with the new changes 🚢 

You should now see checked lists on top in the hover card and in the add/remove modal. Hope this helps!

**Comment by ldexterldesign** [comment]
Sweet, yep that helps!

Kind thanks
